### PR TITLE
Use grafana URL to test receivers

### PR DESCRIPTION
**Pull request by titolins**
Fixes https://github.com/grafana/alerting-squad/issues/872

As described, alerts should use the Grafana URL instead of the remote AM's. For regular alert notifications this was already done, this PR fixes for tests

### CODE REVIEW

File: pkg/alertmanager/alertmanager.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ type Config struct {
 	Retention                         time.Duration
 	MaxConcurrentGetRequestsPerTenant int
 	ExternalURL                       *url.URL
+	TmplExternalURL                   *url.URL
```
Comment (santihernandezc) [inline pkg/alertmanager/alertmanager.go:95]: This new field is not "config" in the same sense as the other fields in the struct, i.e. global config taken from a YAML file and applied to all Alertmanagers.

I think the `Alertmanager` struct would be a better fit for this field (you could add `tmplExternalURL` [here](https://github.com/grafana/mimir/blob/369932771fb4eacef7209b4fd5046965af999513/pkg/alertmanager/alertmanager.go#L130-L131)).

File: pkg/alertmanager/multitenant.go [hunk 2]
```diff
@@ -901,6 +901,7 @@ func (am *MultitenantAlertmanager) newAlertmanager(userID string, amConfig *defi
 		Retention:                         am.cfg.Retention,
 		MaxConcurrentGetRequestsPerTenant: am.cfg.MaxConcurrentGetRequestsPerTenant,
 		ExternalURL:                       am.cfg.ExternalURL.URL,
+		TmplExternalURL:                   tmplExternalURL,
```
Comment (santihernandezc) [inline pkg/alertmanager/multitenant.go:904]: The external URL can change when a new config is applied. For example, consider the following scenario:
1. A user has no Grafana configuration, they are using Mimir's external URL
2. They upload some Grafana configuration, they should be using Grafana's external URL

We should assign the correct external URL on every config sync (check [here](https://github.com/grafana/mimir/blob/369932771fb4eacef7209b4fd5046965af999513/pkg/alertmanager/alertmanager.go#L518-L520), we're updating the templates in the Alertmanager struct, we could use the same lock to update the external URL).

**Review by santihernandezc — CHANGES_REQUESTED** [review]
Nice one! A few comments.

File: pkg/alertmanager/alertmanager.go (hunk 1 above)
Comment (titolins) [reply pkg/alertmanager/alertmanager.go]: Got it! Changed it now 👍

File: pkg/alertmanager/multitenant.go (hunk 2 above)
Comment (titolins) [reply pkg/alertmanager/multitenant.go]: Ah yeah, ofc.. I missed the `else` and presumed a new instance would be built every time. Tks, fixed now 👍

File: pkg/alertmanager/alertmanager.go [hunk 3]
```diff
@@ -191,7 +192,7 @@ type Replicator interface {
 }
 
 // New creates a new Alertmanager.
-func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
+func New(cfg *Config, reg *prometheus.Registry, tmplExternalURL *url.URL) (*Alertmanager, error) {
```
Comment (santihernandezc) [inline pkg/alertmanager/alertmanager.go:195]: If we're already passing the URL in `ApplyConfig()`, we can skip adding a new parameter here. On the initial sync ([right after the Alertmanager is created](https://github.com/grafana/mimir/blob/f52911d917c8c52e0da6a59348a64dd7f7622072/pkg/alertmanager/multitenant.go#L907-L929)), the AM will get the correct URL to be used when testing receivers.

I'd remove this new parameter.

**Review by santihernandezc — APPROVED** [review]
I just tested it, it works as expected, nice job! 🎉 
Please address my comment, once that's fixed you can merge the PR. I don't need to re-review it.

**Pull request merged**